src/goto-instrument/cover_basic_blocks.h

```diff
@@ -130,4 +114,33 @@ class cover_basic_blockst final : public cover_blocks_baset
     block_mapt &block_map);
 };
 
+class cover_basic_blocks_javat final : public cover_blocks_baset
+{
+private:
+  // map block number to first instruction of the block
+  std::vector<goto_programt::const_targett> block_infos;
+  // map java indexes to block indexes
+  std::unordered_map<dstringt, std::size_t, dstring_hash> index_to_block;
```

Comment:
Use `irep_idt` (and `irep_id_hash`), never use `dstringt` directly.

Reply:
Alright if you say so. But may I ask, irep_idt is just a typedef dstringt, is there a reason we keep the two separated?

Reply:
The macro `USE_DSTRING` will make an `irep_idt` be a `dstringt`, otherwise an `irep_idt` is a `std::string`. It's a different question whether disabling `USE_DSTRING` is actually beneficial.

Reply:
I just try commenting out the `#define USE_DSTRING` and cbmc does not compile. So I guess it has not been tested for some time.

Reply:
Would you mind creating an issue, possibly including your build log? Certainly it hasn't been tested in a long time, and it should be properly evaluated whether it makes sense at all anymore. But there doesn't really seem any urgency to do this.

Reply:
I've always hoped this would be fixed as IDEs have much better support for std::string - I thought I had raised an issue but that seems to be all in my head

Reply:
issue created here: https://github.com/diffblue/cbmc/issues/1837